code/procs/helpers.dm

```diff
@@ -2442,8 +2442,9 @@ proc/check_whitelist(var/atom/TA, var/list/whitelist, var/mob/user as mob, var/c
 			TA.reagents.del_reagent(reagent_id)
 			found = 1
 	if (found)
+		if(!user) user = usr // using usr here because some of the procs that call this aren't in a position to know the actual user
 		if (user)
-			boutput(user, "[custom_message]") // haine: done -> //TODO: using usr in procs is evil shame on you
+			boutput(user, "[custom_message]")
 		else if (ismob(TA.loc))
 			var/mob/M = TA.loc
 			boutput(M, "[custom_message]")
```

Comment:
this should be evaluated before usr imo

Reply:
I changed it around a bit, is this any better?

Reply:
Actually, user should probably be checked first, if the calling proc goes through the trouble of sending it.

Reply:
Alright, changed it again, please comment. ^_^